components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java

```diff
@@ -0,0 +1,1323 @@
+
+        doNothing().when(mockRoleManagementService).deleteRole(roleId, tenantDomain);
+        SCIMRoleManager roleManager = new SCIMRoleManager(mockRoleManagementService, tenantDomain);
+        roleManager.deleteRole(roleId);
+        verify(mockRoleManagementService, times(1)).deleteRole(roleId, tenantDomain);
+    }
+
+    @DataProvider(name = "dataProviderForListRolesWithGETSortingNotSupport")
+    public Object[][] dataProviderForListRolesWithGETSortingNotSupport() {
+
+        return new Object[][]{
+                {1, 3, "name", "ascending"},
+                {2, 2, null, "ascending"},
+                {2, 5, "", "ascending"},
+                {0, 0, "name", null},
+                {3, 0, "name", ""},
+                {3, 0, "", ""}
+        };
+    }
+
+    @Test(dataProvider = "dataProviderForListRolesWithGETSortingNotSupport")
+    public void testListRolesWithGETSortingNotSupport(Integer startIndex, Integer count, String sortBy,
+                                                      String sortOrder) {
+
+        SCIMRoleManager roleManager = new SCIMRoleManager(mockRoleManagementService, SAMPLE_TENANT_DOMAIN);
+        assertThrows(NotImplementedException.class, () -> roleManager.
+                listRolesWithGET(null, startIndex, count, sortBy, sortOrder));
+    }
+
+    @Test
+    public void testListRolesWithGETCountNullZero()
+            throws NotImplementedException, BadRequestException, CharonException {
+
+        SCIMRoleManager roleManager = new SCIMRoleManager(mockRoleManagementService, SAMPLE_TENANT_DOMAIN);
+        List<Object> roles = roleManager.listRolesWithGET(null, 1, 0,
+                null, null);
+        assertEquals(roles.size(), 0);
+    }
+
+    @DataProvider(name = "dataProviderForListRolesWithGETInvalidLimit")
+    public Object[][] dataProviderForListRolesWithGETInvalidLimit() {
+
+        return new Object[][]{
+                {"Expression", -2},
+                {null, -5},
+        };
+    }
+
+    @Test(dataProvider = "dataProviderForListRolesWithGETInvalidLimit")
+    public void testListRolesWithGETInvalidLimit(String nodeType, Integer count)
+            throws IdentityRoleManagementException {
+
+        Node rootNode = generateNodeBasedOnNodeType(nodeType, null);
+        when(mockRoleManagementService.getRoles(anyInt(), anyInt(), anyString(), anyString(), anyString())).
+                thenAnswer(invocationOnMock -> {
+                    Integer countArg = invocationOnMock.getArgumentAt(0, Integer.class);
+                    if (countArg != null && countArg < 0) {
+                        String errorMessage =
+                                "Invalid limit requested. Limit value should be greater than or equal to zero. limit: "
+                                        + count;
+                        throw new IdentityRoleManagementClientException(INVALID_LIMIT.getCode(), errorMessage);
```

Comment:
Shall we format this in a nicer way?

Reply:
Fixed https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/387